src/drc/tests/test_partupload.py

```diff
@@ -0,0 +1,272 @@
+        self.canonical.refresh_from_db()
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+        self.assertNotEqual(self.canonical.lock, '')
+
+    def _upload_part_files(self):
+        part_files = split_file(self.file_content, settings.CHUNK_SIZE)
+        for part in self.bestandsdelen:
+            part_url = get_operation_url('bestandsdeel_update', uuid=part.uuid)
+
+            response = self.client.put(
+                part_url,
+                {'inhoud': part_files.pop(0)},
+                format='multipart'
+            )
+
+            self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+            part.refresh_from_db()
+
+            self.assertNotEqual(part.inhoud, '')
+            self.assertEqual(part.voltooid, True)
+
+    def _mark_complete(self):
+        complete_url = get_operation_url('enkelvoudiginformatieobject_complete', uuid=self.eio.uuid)
+
+        response = self.client.put(complete_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK, response.data)
+
+        data = response.json()
+        self.canonical.refresh_from_db()
+        self.eio.refresh_from_db()
+        file_url = get_operation_url('enkelvoudiginformatieobject_download', uuid=self.eio.uuid)
+
+        self.assertEqual(self.canonical.bestandsdelen.count(), 0)
+        self.assertNotEqual(self.eio.inhoud.path, '')
+        self.assertEqual(self.eio.inhoud.size, self.file_content.size)
+        self.assertEqual(data['inhoud'], f'http://testserver{file_url}')
+
+    def _download_file(self):
+        file_url = get_operation_url('enkelvoudiginformatieobject_download', uuid=self.eio.uuid)
+
+        response = self.client.get(file_url)
+
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.content, b'filecontentstring')
+
+    def test_upload_file_full_process(self):
+        self._create_metadata()
+        self._lock_document()
+        self._upload_part_files()
+        self._mark_complete()
+        self._download_file()
+
+    def test_upload_file_wrong_size(self):
+        self._create_metadata()
+        self._lock_document()
+
+        # change file size for part file
+        part = self.bestandsdelen[0]
+        part.grootte = part.grootte + 1
+        part.save()
+
+        # upload part file
+        part_url = get_operation_url('bestandsdeel_update', uuid=part.uuid)
+        part_file = split_file(self.file_content, settings.CHUNK_SIZE)[0]
+
+        response = self.client.put(
+            part_url,
+            {'inhoud': part_file},
+            format='multipart'
+        )
+
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+        error = get_validation_errors(response, 'nonFieldErrors')
+        self.assertEqual(error['code'], 'file-size')
+
+    def test_upload_not_locked(self):
+        self._create_metadata()
+        self._upload_part_files()
```

Comment:
this should not be possible if the document is not locked, I think, even when creating a new file -> if there's no content inline (see "existential questions"), there should probably be a lock ID in the response from the get go.

Reply:
So the current logic is:
* if the document with base64 `inhoud` is created, it won't be locked
* if the document with empty `inhoud` and `bestandsomvang` = 0 (or NULL) is created, it won't be locked
* if the document with empty `inhoud` and `bestandsomvang` > 0 it will be locked right after the creation (and `bestandsdelen` object will be created)